include/lsst/meas/astrom/pessimisticPatternMatcherUtils.h

```diff
@@ -78,48 +127,32 @@ struct ShiftRotMatrixResult {
  *     and reference pair distances to consider the reference pair a candidate
  *     for the source pair. Also sets the tolerance between the opening angles
  *     of the spokes when compared to the reference.
- * @return Struct containing: candidate pairs for this matched pattern; the
- *     rotation matrix to match the patterns, the implied rotations to create
- *     the matrix, a boolean representing success of the algorithm.
+ * @returns The result of the pattern matching.
  */
 PatternResult construct_pattern_and_shift_rot_matrix(
         ndarray::Array<double, 2, 1> src_pattern_array, ndarray::Array<double, 2, 1> src_delta_array,
         ndarray::Array<double, 1, 1> src_dist_array, ndarray::Array<float, 1, 1> dist_array,
-        ndarray::Array<uint16_t, 2, 1> id_array, ndarray::Array<double, 2, 1> reference_array, int n_match,
+        ndarray::Array<uint16_t, 2, 1> id_array, ndarray::Array<double, 2, 1> reference_array, size_t n_match,
```

Comment:
Now I see what you meant about the casts being "unneccessary". While it does make the implementation code cleaner, having the parameter type depend on the implementation is a bad idea in general. The documentation is

> Number of points to attempt to create a pattern from. Must be <= len(src_pattern_array)

which I guess is _sort of_ length-like? 😕

Reply:
Maybe it should be `uint16_t` instead (given that's the array size counter elsewhere in this, though I'm not positive it applies here), but I'm not sure how `parameter type depend[s] on the implementation` here.